packages/adapter-vercel/files/serverless.js

```diff
@@ -22,10 +22,12 @@ export default async (req, res) => {
 		const [path, search] = req.url.split('?');
 
 		const params = new URLSearchParams(search);
-		const pathname = params.get('__pathname');
+		let pathname = params.get('__pathname');
 
 		if (pathname) {
 			params.delete('__pathname');
+			// Optional routes' pathname replacements look like `/foo/$1/bar` which means we could end up with an url like /foo//bar
+			pathname = pathname.replace(/\/+/g, '/');
 			req.url = `${pathname}${path.endsWith(DATA_SUFFIX) ? DATA_SUFFIX : ''}?${params}`;
```

Comment:
Is this safe to do like this? `pathname` sounds like it should not contain the host/origin etc so not the protocol (`https://`) part which we should leave alone, but `req.url` sounds like it _is_ part if it